cmd/fleet/main.go

```diff
@@ -430,8 +434,21 @@ func NewFleetServer(cfg *config.Config, bi build.Info, reporter status.Reporter)
 
 type runFunc func(context.Context) error
 
-// Run runs the fleet server.
+// Run runs the fleet server
 func (f *FleetServer) Run(ctx context.Context) error {
+	// Set up a noop consumer for the ready signal
+	c := make(chan struct{})
+	go func() {
+		for range c {
+		}
+	}()
+	return f.RunWithReadyChannel(ctx, c)
+}
+
+// RunWithReadyChannel runs the fleet server with a ready signal channel
+func (f *FleetServer) RunWithReadyChannel(ctx context.Context, readyCh chan struct{}) error {
```

Comment:
I don't understand this change, we are creating a channel to delay in some way the startup is that correct?

Reply:
when adding the limit calculation in response to a config being passed down there was a race condition introduced with one of the integration tests that read off the fleet input. the existing code doesn't really change in behavior (hopefully), as i converted the `Run` method to use this one and just eat all the signals from the ready channel.

i'm not 100% sure if this is the cleanest way to implement this functionality, it was just the first thing that came to mind.

Reply:
Can we investigate the root cause, the goroutine/channel will eventually be flaky.

Reply:
I updated the PR to protect the test from the race condition by introducing a mutex to the config. currently it only protects against writes in the methods of the config, and doesn't protect against things reaching directly into the config to set things.

This feels like I'm violating some assumption about how the config object is being built and used, since it wasn't threadsafe to begin with. I'm going to dig around to see if I can replace any direct access of the config with some threadsafe getters/setters, but wanted to see what you thought about the approach.